src/zod.ts

```diff
@@ -1,231 +1,211 @@
-/**
- * Zod validation schemas for config.js.
- *
- * All validate functions accept value for configs setting
- */
-
 import { z } from "zod";
 import { Action, LinkType, MatchMode } from "./constants.js";
 import { existsSync } from "fs";
+import { logger } from "./logger.js";
+import { getFileConfig } from "./configuration.js";
+
+const fileConfig = await getFileConfig();
 
 /**
  * VALIDATION_SCHEMA is just a object of all the zod schemas
  * each are named after what they are intended to validate
  */
-export const VALIDATION_SCHEMA = {
-	URL: z.union([z.string(), z.array(z.string())]).refine(
-		(values) => {
-			const validateUrl = (url) => {
-				try {
-					new URL(url);
-					return true;
-				} catch (error) {
-					return false;
-				}
-			};
 
-			if (Array.isArray(values)) {
-				return values.every(
-					(url) => typeof url === "string" && validateUrl(url)
-				);
-			} else {
-				return typeof values === "string" && validateUrl(values);
-			}
-		},
-		{
-			message: "invalid URL",
-		}
-	),
-	PATH: z.union([z.string(), z.array(z.string())]).refine(
-		(values) => {
-			const validatePath = (value) => {
-				if (typeof value !== "string" || !existsSync(value)) {
-					return false;
-				}
-				return true;
-			};
+export const VALIDATION_SCHEMA = z
+	.object({
+		notificationWebhookUrl: z.union([
+			z.undefined(),
```

Comment:
When `undefined` is an option we should probably use `.optional()` rather than an union with undefined